docs/guides/authjs-migration.mdx

```diff
@@ -0,0 +1,373 @@
+
+```tsx filename="app/page.tsx"
+import { auth } from "@clerk/nextjs";
+ 
+export default async function Page() {
+ 
+  const { sessionClaims: { userId } } = auth()
+  console.log(userId)
+
+  return (
+    <p>Home Page</p>
+  )
+
+}
+```
+
+<Callout type="info">
+You can not access the above from the client side. If you are using one of Clerk's hooks then you will need to check if `externalID` exists and if not then read the `userId`. 
+</Callout>
+
+#### Update your database
+
+Alternatively, after the data migration, you can update all the user IDs stored in your database as a foreign key to the new Clerk user IDs.
+
+You can read more about user IDs and user data migration in the [Migration Script README](https://github.com/clerk/migration-script?tab=readme-ov-file#handling-the-foreign-key-constraint).
+
+### Create a production instance
+
+Every Clerk application has a Development and a Production instance. Before you start migrating user data, you will configure production instance and migrate your users directly into that instance. The [Deploying to Production](/docs/deployments/overview#deploying-to-production) page covers creating a production instance.
+
+You can migrate a small set of users on the development instance for testing/staging. To enable importing users to your development instance, add `IMPORT_TO_DEV_INSTANCE=true` to the `.env` for the migration script.
+
+### Migrate User Data
+
+This walkthrough will help you move user data from your existing database to Clerk.
+
+To retain the user data in your database for easy querying, check out the documentation on [data synchronization with webhooks](/docs/users/sync-data).
+
+1. Clone `github.com/clerk/migration-script`
+2. Create a `.env` file in the root of the cloned repository with the `CLERK_SECRET_KEY` **of your production instance**.
+3. Export all the user data from your database into a users.json file. The file should be in the following format.
+```json filename="users.json"
+[
+  {
+    "userId": "string",
+    "email": "email",
+    "firstName": "string (optional)",
+    "lastName": "string (optional)",
+    "password": "string (optional)",
+    "passwordHasher": "argon2 | argon | bcrypt | md5 | pbkdf2_sha256 | pbkdf2_sha256_django | pbkdf2_sha1 | scrypt_firebase",
+  }
+]
+```
+4. If you already have an API endpoint in your Auth.js app that returns a list of users you could use that. Otherwise you will need to query your database to obtain the user information, or use an export function from a database management tool.
+    
+```sql
+  SELECT id as userId, email, name FROM users
+```
+    
+5. Edit the `.env.` file in the migration, and add your Clerk Secret Key using `CLERK_SECRET_KEY`
+6. Run the script with `npm start`
+7. Verify by signing in to your application secured by Clerk, and by checking the users listed in the [Users](https://dashboard.clerk.com/last-active?path=users) section of the Clerk Dashboard.
```

Comment:
I think this is what you mean? The user should sign in to their instance (creating their user) and then they can check the dashboard that their user successfully exists as a user of the instance??

```suggestion
7. Verify the migration to Clerk was successful. Sign in to your application secured by Clerk. Then, check that your user is listed in the [Users](https://dashboard.clerk.com/last-active?path=users) section of the Clerk Dashboard.
```

Reply:
It doesn't need to be in either order. I will probably change it to check the Dashboard first as that makes more sense.